### PR TITLE
Improve and amend readme

### PR DESCRIPTION
**Pull request by raproid**
What I did: 
I improved the README file.

How I did it: 
- Added a more comprehensive description
- Improved grammar overall
- Added more examples of usage in a user's code
- Added the company logo and "Contribute", "Contacts" and "License" section.
- More things that don't come to mind at the moment

How can you check it:
Compare the improved README against the master README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,50 +1,88 @@
-# insconfig
+[<img src="https://github.com/insolar/doc-pics/raw/master/st/github-readme-banner.png">](http://insolar.io/?utm_source=Github)
+
+# Insolar configurations
 
 ![test](https://github.com/insolar/insconfig/workflows/test/badge.svg)
 
-Config management library.
-This is the wrapper on https://github.com/spf13/viper library
+Insolar configurations unify configuring for all Insolar applications: [Assured Ledger](https://github.com/insolar/assured-ledger), [Insolar Observer](https://github.com/insolar/observer), [Insolar MainNet](https://github.com/insolar/mainnet), [Insolar Explorer](https://github.com/insolar/block-explorer) and more.
```
Comment (sofiayem) [inline README.md:7]: What more?

File: README.md [hunk 2]
```diff
@@ -1,50 +1,88 @@
...
+- Environment values override file values.
+- Option to use only ENV, without a configuration file at all.
+- Option to write the config file content to the log file at an app launch. 
+- Hiding sensitive data via the `insconfigsecret` tag.
+- No default values in a configuration file. All values should be set explicitly, otherwise the library returns an error.
+- No unnecessary field or parameters both in a configuration file and ENV, otherwise the library returns an error. Consider as unecessary: fields in a config struct unused in a configuration file, old or obsolete parameters in a configuration file that are not currently used, unused parameters in ENV.
+- Support of custom flags, go flags and pflags.
+- No overriding configutation files by flags.
+- Option to generate an empty .yaml file with field descriptions.
+- Automatic adding of the `--config` flag
```
Comment (sofiayem) [inline README.md:27]: Full stop?

File: README.md [hunk 3]
```diff
@@ -1,50 +1,88 @@
...
+- [work in progress] Automatic of the `--gen-config` flag
```
Comment (sofiayem) [inline README.md:28]: Automatic _adding_ ... full stop?

File: README.md [hunk 4]
```diff
@@ -57,7 +95,9 @@ With custom spf13/pflags
     fmt.Println(testflag1)
 ```
 
-With spf13/cobra. Cobra doesn't provide tools to manage flags parsing, so you need to add config flag yourself
+#### Custom [spf13/cobra flags](https://github.com/spf13/cobra). 
+
+Note: Cobra doesn't provide tools for managing flags parsing, so you need to add the `--config flag yourself.
```
Comment (sofiayem) [inline README.md:100]: - that manage flag parsing?
- you forgot to close the in-line code-snippet (--config) with the `

File: README.md [hunk 5]
```diff
@@ -82,42 +122,83 @@ func main () {
...
     }
     ...
-    insconfig.NewYamlTemplater(new(Configuration)).TemplateTo(os.StdOut)
+    insconfig.NewYamlTemplater(new(Config)).TemplateTo(os.StdOut)
 ```
 
-### Dump config with hidden fields
+### Generating a configuration template with hidden fields
+
+If you have some sensitive data you may want to hide it in a config. You can use the `insconfigsecret` tag to hide such data.
```
Comment (sofiayem) [inline README.md:155]: Forgot a comma: If you have some sensitive data, ...

File: README.md (hunk 1 above)
Comment (raproid) [reply README.md:7]: More Insolar applications. I don't know if we currently have more. This is a general statement for the future.

File: README.md (hunk 2 above)
Comment (raproid) [reply README.md]: fixed.

File: README.md (hunk 4 above)
Comment (raproid) [reply README.md:100]: closed the code snipped.
"for doing something" or "that do something" weigh the same here. You use more "that do something"?

File: README.md (hunk 5 above)
Comment (raproid) [reply README.md:155]: fixed. tnx

File: README.md (hunk 4 above)
Comment (raproid) [reply README.md]: that manage flag parsing

**Pull request merged**